ecl/hql/hqlgram.y

```diff
@@ -10560,6 +10566,46 @@ optSortList
     | ',' sortList
     ;
 
+optTraceFlags
+    : traceFlags
+    |                   {   $$.setNullExpr(); }
+    ;
+
+traceFlags
+    : ',' traceFlag     {   $$.setExpr($2.getExpr()); }
+    | traceFlags ',' traceFlag
+                        {   $$.setExpr(createComma($1.getExpr(), $3.getExpr())); }
+    ;
+
+traceFlag
+    : KEEP expression   {
+                            parser->normalizeExpression($2, type_int, false);
+                            $$.setExpr(createExprAttribute(keepAtom, $2.getExpr()));
```

Comment:
Worth adding an extra $1 parameter to the setExpr() call to set the associated position